src/python/antares_xpansion/input_checker.py

```diff
@@ -292,50 +427,62 @@ def check_setting_option_type(option, value):
                  False or exits if the value has the wrong type
     """
 
-    options_types = {
-        "method": "string",
-        "uc_type": "string",
-        "master": "string",
-        "optimality_gap": "double",
-        "relative_gap": "double",
-        "cut_type": "string",
-        "week_selection": "string",
-        "max_iteration": "integer",
-        "relaxed_optimality_gap": "string",
-        "solver": "string",
-        "timelimit": "integer",
-        "yearly-weights": "string",
-        "additional-constraints": "string",
-        "log_level": "integer",
-    }
-    option_type = options_types.get(option)
-    if option_type is None:
+    if options_types_and_legal_values.get(option) is None:
         flushed_print(
             'check_setting_option_type: Illegal %s option in settings file.' % option)
-        sys.exit(1)
-    else:
-        if option_type == 'string':
+        raise NotHandledOption
+
```

Comment:
Introduce a warning for deprecated options similarly as what is done for the check of candidates (or even raise an exception, easier to do as we just need remove deprecated options from `option_types_and_legal_values`) -> need to be discussed with Mathilde